src/OCServo.cpp

```diff
@@ -0,0 +1,481 @@
+    byte params[2] = {
+        lowByte(voltage),
+        highByte(voltage)
+    };
+    return this->ocsWrite(OCS_MAX_VOLTAGE, 2, params);
+}
+
+OCSResponse OCServo::setMinVoltage(int voltage) {
+    byte params[2] = {
+        lowByte(voltage),
+        highByte(voltage)
+    };
+    return this->ocsWrite(OCS_MIN_VOLTAGE, 2, params);
+}
+
+OCSResponse OCServo::setOperationSpeed(long speed) {
+    byte params[2] = {
+        lowByte(speed),
+        highByte(speed)
+    };
+    return this->ocsWrite(OCS_OPERATION_SPEED, 2, params);
+}
+
+/* READ COMMANDS */
+byte OCServo::getID() {
+    OCSResponse response = this->ocsRead(OCS_SERVO_ID, 1);
+    if (response.numberOfParameters != 1) {
+        return 0xFF;
+    }
+    return response.parameters[0];
+}
+
+long OCServo::getBaudRate() {
+    OCSResponse response = this->ocsRead(OCS_BAUD_RATE, 1);
+    if (response.numberOfParameters != 1) {
+        return -1;
+    }
+    return this->byteToBaudRate(response.parameters[0]);
+}
+
+int OCServo::getMaxTorque() {
+    OCSResponse response = this->ocsRead(OCS_MAX_TORQUE, 2);
+    if (response.numberOfParameters != 2) {
+        return -1;
+    }
+    return ((response.parameters[1] << 8) | response.parameters[0] & 0x00FF);
+}
+
+int OCServo::getMode() {
+    OCSResponse response = this->ocsRead(OCS_RUNNING_MODE, 1);
+    if (response.numberOfParameters != 1) {
+        return -1;
+    }
+    return response.parameters[0];
+}
+
+int OCServo::getGoalAngle() {
+    OCSResponse response = this->ocsRead(OCS_GOAL_POSITION, 2);
+    if (response.numberOfParameters != 2) {
+        return -1;
+    }
+    int value = ((response.parameters[1] << 8) | response.parameters[0] & 0x00FF);
+    return map(value, 0, 4095, 0, 360);
+}
+
+int OCServo::getGoalPosition() {
+    OCSResponse response = this->ocsRead(OCS_GOAL_POSITION, 2);
+    if (response.numberOfParameters != 2) {
+        return -1;
+    }
+    int value = ((response.parameters[1] << 8) | response.parameters[0] & 0x00FF);
+    return value;
+}
+
+long OCServo::getOperationTime() {
+    OCSResponse response = this->ocsRead(OCS_OPERATION_TIME, 2);
+    if (response.numberOfParameters != 2) {
+        return -1;
+    }
+    return ((response.parameters[1] << 8) | response.parameters[0] & 0x00FF);
+
+}
+
+int OCServo::getResponseDelay() {
+    OCSResponse response = this->ocsRead(OCS_RESPONSE_DELAY, 1);
+    if (response.numberOfParameters != 1) {
+        return -1;
+    }
+    return response.parameters[0] * 2;
+}
+
+int OCServo::getResponseLevel() {
+    OCSResponse response = this->ocsRead(OCS_RESPONSE_LEVEL, 1);
+    if (response.numberOfParameters != 1) {
+        return -1;
+    }
+    return response.parameters[0];
+}
+
+int OCServo::getMinAngle() {
+    OCSResponse response = this->ocsRead(OCS_MIN_ANGLE, 2);
+    if (response.numberOfParameters != 2) {
+        return -1;
+    }
+    return ((response.parameters[1] << 8) | response.parameters[0] & 0x00FF);
```

Comment:
make a define for it, it is easy to make typo mistake

`#define BYTES_TO_UINT16(high, low) (((high) << 8) | ((low) & 0xFF))`

```suggestion
    return BYTES_TO_UINT16(response.parameters[1], response.parameters[0]);
```